install/terraform/cluster.tf

```diff
@@ -0,0 +1,177 @@
+# Copyright 2019 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+provider "google-beta" {
+  version = "~> 2.10"
+  zone    = "${var.cluster["zone"]}"
+}
+
+provider "google" {
+  version = "~> 2.10"
+}
+
+
+# Ports can be overriden using tfvars file
+variable "ports" {
+  default = "7000-8000"
+}
+
+# Set of GKE cluster parameters which defines its name, zone
+# and primary node pool configuration.
+# It is crucial to set valid ProjectID for "project".
+variable "cluster" {
+  description = "Set of GKE cluster parameters."
+  type        = "map"
+
+  default = {
+    "zone"             = "us-west1-c"
+    "name"             = "test-cluster"
+    "machineType"      = "n1-standard-4"
+    "initialNodeCount" = "4"
+    "project"          = "agones"
+  }
+}
+
+# echo command used for debugging purpose
+# Run `terraform taint null_resource.test-setting-variables` before second execution
+resource "null_resource" "test-setting-variables" {
+  provisioner "local-exec" {
+    command = "${"${format("echo Current variables set as following - name: %s, project: %s, machineType: %s, initialNodeCount: %s, zone: %s",
+      "${var.cluster["name"]}", "${var.cluster["project"]}",
+      "${var.cluster["machineType"]}", "${var.cluster["initialNodeCount"]}",
+    "${var.cluster["zone"]}")}"}"
+  }
+}
+
+resource "google_container_cluster" "primary" {
```

Comment:
This seems to be missing `min_master_version = "1.13"` so that we create clusters at the desired version.